src/Symfony/Component/Validator/Constraints/Compound.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * Extend this class to create a reusable set of constraints.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+abstract class Compound extends Composite
+{
+    /** @var Constraint[] */
+    public $constraints = [];
+
+    public function __construct($options = null)
+    {
+        if (isset($options[$this->getCompositeOption()])) {
+            throw new ConstraintDefinitionException(sprintf('You can\'t redefine the "%s" option. Use the %s::getConstraints() method instead.', $this->getCompositeOption(), __CLASS__));
+        }
+
+        if (0 === \count($constraints = $this->getConstraints($this->normalizeOptions($options)))) {
+            throw new ConstraintDefinitionException(sprintf('The %s::getConstraints() method must return at least one constraint.', __CLASS__));
```

Comment:
why? can't it make sense to have zero constraints when they are generated from somewhere else?

Reply:
As this feature consists of the developper writing a custom constraint extending `Compound`, I considered providing no constraint at all is a logic issue.

> when they are generated from somewhere else

What do you have in mind? I don't see cases where such constraints would be generated.

Reply:
The same could be said about having just one constraint. But that'd be artificial.

Are there any technical reasons to reject the zero-constraints case? If not, we should allow it. Otherwise, we're forcing ppl that have a use case to add boilerplate code just for working around an artificial limitation of the code. *Because* we can't imagine when this is useful, then we need to not forbid it when it doesn't conflict with some technicalities.

Reply:
Deal